python/pyiceberg/catalog/__init__.py

```diff
@@ -25,6 +26,41 @@
 from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
 from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
 from pyiceberg.typedef import EMPTY_DICT, Identifier, Properties
+from pyiceberg.utils.config import Config, merge_config
+
+logger = logging.getLogger(__name__)
+
+_env_config = Config()
+
+
+def load_catalog(name: str, uri: str, **properties: str | None) -> Catalog:
```

Comment:
Can we pass `uri` as a property instead of explicitly? Not all catalogs use a URI. Dynamo, for example, just uses a table name.

Reply:
Of course, updated! Keep in mind that we determine the catalog based on the `uri`, so for dynamo you should set the URI to Dynamo. We could also pass the properties to a static method on the Catalog, which then returns a boolean if the properties match the catalog.